graylog2-server/src/main/java/org/graylog2/database/utils/MongoUtils.java

```diff
@@ -238,4 +244,24 @@ public T save(BuildableMongoEntity<T, ?> entity) {
             return orig;
         }
     }
+
+    /**
+     * Counts the number of documents for each distinct value of the given field.
+     *
+     * @param field the field to group by.
+     * @return Map of field values to their respective counts.
+     */
+    public Map<String, Long> countByField(String field) {
+        final Map<String, Long> counts = new HashMap<>();
+        collection.aggregate(
+                List.of(Aggregates.group("$" + field, Accumulators.sum("count", 1))),
+                Document.class
+        ).forEach(doc -> {
+            Object id = doc.get("_id");
```

Comment:
Not entirely sure, but can't you treat this as String-field directly?
```suggestion
            String id = doc.getString("_id");
```
Then you don't have to call `.toString()` in the line below.